src/mlpack/methods/lsh/lsh_search.hpp

```diff
@@ -188,7 +210,7 @@ class LSHSearch
    * are private members of this class, initialized during the class
    * initialization.
    */
-  void BuildHash();
+  void BuildHash(const std::vector<arma::mat> &projection);
```

Comment:
If you set `LSHSearch::projections` in either the constructor or in `Train()` then there is no need to pass this parameter (or check its validity inside of `BuildHash()`).  Actually to be honest I would be happy to move `BuildHash()` entirely into `Train()` since `BuildHash()` is only called once, but either way is fine.